platform/engineering/design-docs/2020-06-02-Sentry10.md

```diff
@@ -0,0 +1,142 @@
+sentry_onpremise_symbolicator-cleanup_1     /entrypoint.sh 55 23 * * * ...   Up      3021/tcp                    
+sentry_onpremise_symbolicator_1             /bin/bash /docker-entrypoi ...   Up      3021/tcp                    
+sentry_onpremise_web_1                      /bin/sh -c exec /docker-en ...   Up      0.0.0.0:9000->9000/tcp      
+sentry_onpremise_worker_1                   /bin/sh -c exec /docker-en ...   Up      9000/tcp                    
+sentry_onpremise_zookeeper_1                /etc/confluent/docker/run        Up      2181/tcp, 2888/tcp, 3888/tcp
+```
+
+For comparison sake, our Sentry 9 uses 3 docker containers:
+
+- web
+- worker
+- cron
+
+
+### Code Location
+All code will live in the [`devops` repo](https://github.com/department-of-veterans-affairs/devops)
+
+### Testing Plan
+We will stand up Sentry 10 in parallel with our existing Sentry 9.1.2. We will then configure the `Settings.sentry.dsn` of `dev` and/or `staging` to use the Sentry 10 deployment.
+
+### Logging
+N/A
+
+### Debugging
+N/A - I think?
+
+### Caveats
+_To be determined._
+
+### Security Concerns
+No additional security concerns to the ones we have with our existing Sentry.
+
+### Privacy Concerns
+No additional privacy concerns to the ones we have with our existing Sentry.
+
+### Open Questions and Risks
+#### 1. Performance Scaling
+Since we intend on deploying to a single EC2 instance, there is not much room for scaling up in the event that Sentry 10 runs into CPU or memory bottlenecks.
+
+#### 2. Moving Away from `getsentry/onpremise` Happy Path
+The [`getsentry/onpremise` repo](https://github.com/getsentry/onpremise) is a great way to quickly run your own "on premise" Sentry 10 application. It was not designed with scale in mind. In this design, we will be heavily relying on this bootstrapping repo for building and deploying. It's likely that we will eventually have a need to do something custom that would require us to fork the `getsentry/onpremise` repo, making any future modifications and upgrades difficult.
+
+### Work Estimates
+#### Phase 1 - Ansible Build
+Build the AWS AMI. In this step, all docker images should be downloaded onto the machine. This would be considered the ansible "build".
+
+#### Phase 2a - Ansible Deployment
+Create the ansible code to deploy the Sentr10 AMI. This might just be a deployment configuration or it might require new deployment ansible roles or modification to existing roles.
+
+#### Phase 2b - AWS Deplyoment Infra (terraform)
+Create the necessary terraform code to build the AWS infrastructure needed for deploying into our `vagov-utility` VPC.
+
+:warning: There's a bidirectional dependency on the ansible deployment scripts & aws infra needed - sometimes you don't know which infrastructure is needed until you have the deployment config and sometimes you don't know which deployment configuration is needed until you know what infrastrucutre is available. :warning:
+
+
+#### Phase 3 - Observe, Modify, Iterate
```

Comment:
Phase 3 would have some form of cutover or migration of traffic?  Apps  would need to change SENTRY_DSN at least?  Is there a chunk of time needed to setup GitHub integration?

Reply:
I wrote a quick blip about the dsn config in ["Testing Plan" section](https://github.com/department-of-veterans-affairs/va.gov-team/pull/9725/files#diff-5cbfe5b3cf8d48b44068de25a8187603R78), I'll expand on it a bit more.